### PR TITLE
Feature/extract export type

### PR DESCRIPTION
**Pull request by KaneCheshire**
Updated to extract the export type from the gym export options plist, given every project now needs one of those. Also tries to extract the team ID from the plist.

### CODE REVIEW

File: Fastfile [hunk 1]
```diff
@@ -140,10 +134,34 @@ def _build_with_gym()
...
 end
 
+def _get_export_method()
+  export_method = "enterprise"  # Default export method is `enterprise` since this is the most commonly used
+  if ENV['GYM_EXPORT_OPTIONS'] != nil
+    export_method = get_info_plist_value(path: ENV['GYM_EXPORT_OPTIONS'], key: "method")
+  else
+    UI.message("Falling back to enterprise `export_method` since `GYM_EXPORT_OPTIONS` is not defined")
+  end
+  export_method
```
Comment (KaneCheshire) [inline Fastfile]: Is this actually required at the end of the function in ruby to return it?

Comment (neilkimmett) [reply Fastfile]: Yup. In Ruby people tend to write functions that are as short as possible, frequently just one line with at most one conditional branch. So this function would be 
```ruby
def _get_export_method()
  ENV['GYM_EXPORT_OPTIONS'] != nil ? get_info_plist_value(path: ENV['GYM_EXPORT_OPTIONS'], key: "method") : "enterprise"
end
```
but alas then you lose the ability to do side-effecty things like `UI.message`. So I think this is fine as is.

File: Fastfile [hunk 2]
```diff
@@ -140,10 +134,34 @@ def _build_with_gym()
   end
 end
 
+def _get_export_method()
+  export_method = "enterprise"  # Default export method is `enterprise` since this is the most commonly used
+  if ENV['GYM_EXPORT_OPTIONS'] != nil
```
Comment (neilkimmett) [inline Fastfile]: It might be more idiomatic to use `.present?` instead of `!= nil`

**Review by neilkimmett — APPROVED** [review]

File: Fastfile (hunk 2 above)
Comment (KaneCheshire) [reply Fastfile]: Can also do `.nil?` as well?

File: Fastfile (hunk 1 above)
Comment (NikolaKirev) [reply Fastfile]: What if you have a separate function `_fallback_to_enterprise_export_method()` that does the `UI.message` and returns the default?
That way we can keep the one liner here that @neilkimmett suggests.

File: Fastfile (hunk 2 above)
Comment (neilkimmett) [reply Fastfile]: Yup.

File: Fastfile [hunk 3]
```diff
@@ -140,10 +134,33 @@ def _build_with_gym()
   end
 end
 
+def _get_export_method()
+  ENV['GYM_EXPORT_OPTIONS'].nil? ? _fallback_to_enterprise_export_method() : get_info_plist_value(path: ENV['GYM_EXPORT_OPTIONS'], key: "method")
```
Comment (neilkimmett) [inline Fastfile:138]: Usually you'd omit the `()` if the method takes no params. Same on method declaration. But given we do `()` everywhere else in the file it seems fine.

File: Fastfile (hunk 1 above)
Comment (KaneCheshire) [reply Fastfile]: @NikolaKirev updated

File: Fastfile (hunk 3 above)
Comment (KaneCheshire) [reply Fastfile:138]: @neilkimmett Yeah that's what I thought as well. Could be a nice PR to tidy that up separately though

File: Fastfile [hunk 4]
```diff
@@ -140,10 +134,33 @@ def _build_with_gym()
...
+  UI.message("Falling back to enterprise `export_method` since `GYM_EXPORT_OPTIONS` is not defined")
+  "enterprise"
+end
+
 def _update_team_id_if_necessary()
-  if !ENV['FL_PROJECT_SIGNING_PROJECT_PATH'].to_s.strip.empty? && !ENV['FL_PROJECT_TEAM_ID'].to_s.strip.empty?
-    update_project_team
+  project_path = ENV['FL_PROJECT_SIGNING_PROJECT_PATH']
+  team_id = _get_team_id()
+  if !project_path.to_s.strip.empty? && !team_id.to_s.strip.empty?
```
Comment (NikolaKirev) [inline Fastfile]: I think the double negation here makes the code a bit unreadable.
Is it possible to go for something like:
`project_path.to_s.strip.present? && team_id.to_s.strip.present?`

or we do:
`project_path.to_s.strip.empty? || team_id.to_s.strip.empty?`
and put the "success" flow in the after the `else`

Comment (KaneCheshire) [reply Fastfile]: Prefer the former. Nice! Will update

Comment (KaneCheshire) [reply Fastfile]: @NikolaKirev updated

**Review by NikolaKirev — APPROVED** [review]

**Comment by KaneCheshire** [comment]
I'll delete the branch after I've released 3.0.0 or it'll affect our builds (we're pointing to this branch atm)

**Pull request merged**